metricflow/plan_conversion/time_spine.py

```diff
@@ -34,26 +33,42 @@ def spine_table(self) -> SqlTable:
         return SqlTable(schema_name=self.schema_name, table_name=self.table_name, db_name=self.db_name)
 
     @staticmethod
-    def create_from_manifest(semantic_manifest: SemanticManifest) -> TimeSpineSource:
+    def create_from_manifest(semantic_manifest: SemanticManifest) -> Dict[TimeGranularity, TimeSpineSource]:
         """Creates a time spine source based on what's in the manifest."""
-        time_spine_table_configurations = semantic_manifest.project_configuration.time_spine_table_configurations
-
-        if not (
-            len(time_spine_table_configurations) == 1
-            and time_spine_table_configurations[0].grain == DEFAULT_TIME_GRANULARITY
-        ):
-            raise NotImplementedError(
-                f"Only a single time spine table configuration with {DEFAULT_TIME_GRANULARITY} is currently "
-                f"supported. Got:\n"
-                f"{mf_pformat(time_spine_table_configurations)}"
+        time_spine_sources = {
+            time_spine.primary_column.time_granularity: TimeSpineSource(
+                schema_name=time_spine.node_relation.schema_name,
+                table_name=time_spine.node_relation.relation_name,  # is relation name the table name? double check
+                db_name=time_spine.node_relation.database,
+                time_column_name=time_spine.primary_column.name,
+                time_column_granularity=time_spine.primary_column.time_granularity,
             )
+            for time_spine in semantic_manifest.project_configuration.time_spines
+        }
 
-        time_spine_table_configuration = time_spine_table_configurations[0]
-        time_spine_table = SqlTable.from_string(time_spine_table_configuration.location)
-        return TimeSpineSource(
-            schema_name=time_spine_table.schema_name,
-            table_name=time_spine_table.table_name,
-            db_name=time_spine_table.db_name,
-            time_column_name=time_spine_table_configuration.column_name,
-            time_column_granularity=time_spine_table_configuration.grain,
-        )
+        # For backward compatibility: if legacy time spine config exists in the manifest, add that time spine here for
+        # backward compatibility. Ignore it if there is a new time spine config with the same granularity.
+        legacy_time_spines = semantic_manifest.project_configuration.time_spine_table_configurations
+        for legacy_time_spine in legacy_time_spines:
+            if not time_spine_sources.get(legacy_time_spine.grain):
```

Comment:
We should use `in` instead of `.get()` here:

`if legacy_time_spine.grain not in time_spine_sources:`

That way you don't get weird bugs with falsey hashable values.